mmv1/third_party/terraform/utils/config.go.erb

```diff
@@ -226,7 +227,7 @@ func (c *Config) LoadAndValidate(ctx context.Context) error {
 
 	// 4. Header Transport - outer wrapper to inject additional headers we want to apply
 	// before making requests
-	headerTransport := newTransportWithHeaders(retryTransport)
+	headerTransport := newTransportWithHeaders(retryTransport, c.RequestReason)
```

Comment:
you should be able to change this to 
```suggestion
	headerTransport := newTransportWithHeaders(retryTransport)
	if c.RequestReason != "" {
	  headerTransport.Set("X-Goog-Request-Reason", c.RequestReason)
	}
```

Reply:
@rileykarson

Reply:
yeah, that'd work fine as well.  personally i prefer the clear setting of the header value vs setting it as `headerTransport.Set(`  (which at first glance looks like setting the transport (not header).    eitherway, can make this change.  upto @rileykarson

Reply:
fair point

Reply:
`headerTransport` is a transport that sets headers, similar to how `loggingTransport` is a transport that adds logging and `retryTransport` a transport that adds http retries. Using the `headerTransport` is preferable, I think.

Reply:
to clarify, you want it to look like

```golang
	headerTransport := newTransportWithHeaders(retryTransport)
	if c.RequestReason != "" {
	  headerTransport.Set("X-Goog-Request-Reason", c.RequestReason)
	}
```

Reply:
Seems right to me!

Reply:
added